### PR TITLE
Add test to cover LIBSSH2_ERROR_EAGAIN on ssh_screen module

### PR DESCRIPTION
**Pull request by Wabri**
Related to: https://progress.opensuse.org/issues/174256

### CODE REVIEW

File: consoles/ssh_screen.pm [hunk 1]
```diff
@@ -83,6 +83,8 @@ sub type_string ($self, $nargs) {
             select(undef, undef, undef, 0.1);
         } elsif ($chunk < 0) {
             # Old Net::SSH2 error signaling
+            print "$chunk\n";
+            print LIBSSH2_ERROR_EAGAIN"\n";
```
Comment (Martchus) [inline consoles/ssh_screen.pm:87]: This is a debugging leftover, right?

Comment (Wabri) [reply consoles/ssh_screen.pm:87]: Yes, I need to do a refactor of the code and the commits because they are really bad

**Review by okurz — APPROVED** [review]

**Review by okurz — CHANGES_REQUESTED** [review]
please make sure to catch all log output during tests, see https://github.com/os-autoinst/os-autoinst/actions/runs/12428291481/job/34699625619?pr=2599#step:3:435


```
3: [08:47:30] t/26-serial_screen.t ..................... ok      333 ms ( 0.00 usr  0.00 sys +  0.28 cusr  0.06 csys =  0.34 CPU)
3: [2024-12-20T08:47:31.376293Z] [debug] [pid:2361] <<< consoles::ssh_screen::type_string(text="test")
3: [08:47:32] t/26-ssh_screen.t ........................ ok     1086 ms ( 0.01 usr  0.00 sys +  0.67 cusr  0.11 csys =  0.79 CPU)
```

File: t/26-ssh_screen.t [hunk 2]
```diff
@@ -21,12 +22,30 @@ is($screen->{fd_write}, 'My_Chan', 'SSH channel is used for writing');
...
 };
 
+subtest 'test old net ssh2 error handling' => sub {
+    my $mock_screenconsole = Test::MockModule->new('consoles::serial_screen');
+    my $mock_connection = Test::MockObject->new();
+    my $mock_write_attempts = 0;
+    my $mock_channel = Test::MockObject->new();
+    $mock_channel->mock('write', sub {
+            $mock_write_attempts++;
+            return -37 if $mock_write_attempts <= 3;
```
Comment (kalikiana) [inline t/26-ssh_screen.t:40]: What's this -37? Can we have the magic number here?

Comment (perlpunk) [reply t/26-ssh_screen.t:40]: You can just use the constant exported by the module:
```suggestion
            return LIBSSH2_ERROR_EAGAIN if $mock_write_attempts <= 3;
```

**Comment by okurz** [comment]
you need to tidy, see https://github.com/os-autoinst/os-autoinst/actions/runs/12433234130/job/34714312556?pr=2599#step:4:305

**Review by okurz — CHANGES_REQUESTED** [review]
something went wrong with your commits. Please provide distinct changes in separate commits, no merge commits

File: codecov.yml [hunk 3]
```diff
@@ -21,13 +21,13 @@ coverage:
           - consoles/console.pm
           - consoles/VMWare.pm
           - consoles/VNC.pm
-          - consoles/video_base.pm
```
Comment (perlpunk) [inline codecov.yml:24]: I guess this was an accident :)

Comment (Wabri) [reply codecov.yml]: Yep, of course :grin:

**Review by okurz — APPROVED** [review]

**Review by b10n1k — APPROVED** [review]
LGTM

**Pull request merged**